src/main/kotlin/se/uulm/snowballr/backend/service/ProjectPaperService.kt

```diff
@@ -241,4 +247,25 @@ class ProjectPaperService(
 
             projectPaper.toGrpcProjectPaperWithData(paper)
         }
+
+    override suspend fun getNextPaper(request: Base.Id): GrpcProjectPaper = withUser(userRepo) { currentUser ->
+        val projectPaperId = parseUUID(request.id, EntityType.PROJECT_PAPER)
+        val projectPaper = repo.getProjectPaperById(projectPaperId).getOrThrow()
+        val projectId = projectRepo.getProjectById(projectPaper.projectId).getOrThrow().id
+
+        isAllowedToReadProject(projectMemberRepo).checkFor(currentUser, projectId)
```

Comment:
```suggestion
        val projectPaperId = parseUUID(request.id, EntityType.PROJECT_PAPER)
        
        isAllowedToReadProject(projectMemberRepo).checkFor(currentUser, projectId)
        
        val projectPaper = repo.getProjectPaperById(projectPaperId).getOrThrow()
        val projectId = projectRepo.getProjectById(projectPaper.projectId).getOrThrow().id
```

This way we first check if the user is allowed to access before we access.

Reply:
I also wanted to implement it like this, but this is not possible, since we have to get the project id of the given project paper. But you are right, we could do the check at least before we access the project.

Reply:
I think we do have another issue now. The user is now able to find out which project papers are in a project and which are not. In my opinion, we would need an additional access check before we fetch the project paper. But since we can only decide if the user is allowed to access the project paper, based on the fact, if the is allowed to read the project, we cannot decide that before we fetch the project paper in order to get the id for the project.

I'm not sure how to solve this problem, without completely changing the API. I think we should discuss this tomorrow in our meeting.

Reply:
Yes, I know this is a problem. Me and @Merseleo also had a talk about this. For the moment we agreed to only check the accessibility as good as possible, because otherwise we would have to rework a lot of stuff, which is not efficient regarding the current time managment. But let's talk about this tomorrow.